### ISSUE TITLE
RUST-569 Client Side Field Level Encryption Support

### ISSUE THREAD

**Issue by usefulalgorithm** [mongodb/mongo-rust-driver#520]
Hi, is the encryption feature present in the driver? Looked at the docs and it seemed the Rust driver does not support encryption (both automatic and explicit) for the time being...

Thanks!

**Comment by kmahar** [comment]
Hi @usefulalgorithm, you are correct that we don't currently support this in the driver. The Jira ticket to track this work is: https://jira.mongodb.org/browse/RUST-119

**Comment by masterbater** [comment]
Hello is there an ETA for this feature?

**Comment by LuisOsta** [comment]
@kmahar The ticket you posted there is closed https://jira.mongodb.org/browse/RUST-119. But it doesn't seem that Client FLE support has been added - https://www.mongodb.com/docs/manual/core/csfle/reference/compatibility/#std-label-csfle-compatibility-reference. Is this going to be released yet?

**Comment by kmahar** [comment]
Hi @LuisOsta, sorry for the unclear Jira trail. We closed RUST-119 as a duplicate in favor of an epic with more granular tickets tracking all of the pieces of work involved, RUST-569. I'll update the issue title here to reflect the new ticket number.

We're actively working through implementing CSFLE and queryable encryption support right now (#717 is the latest PR). It's quite a large effort and so I can't provide an exact ETA at this time, but it should be released sometime in the coming months.

**Comment by LuisOsta** [comment]
Thank you for the update! I completely understand that CSFLE is a really large endeavor and I really appreciate the updates. I'll be excited for whenever the features are available in a future release.

**Comment by LuisOsta** [comment]
@kmahar Congrats on the 2.4 release!! I saw that in the release notes https://github.com/mongodb/mongo-rust-driver/releases, it says that the encryption functionality is released and will be unstable for the time being. So y'all have a sense of when it'd be stabilized?

And would you want to close this issue now since its released or wait until its stable and generally usable?

**Comment by JamesKovacs** [comment]
Hi, @LuisOsta,

Thank you for reaching out to clarify. The API for Field-Level Encryption (FLE) is stable and has been present in our other drivers for years. The API for Queryable Encryption (QE) is new and the server functionality is in public technical preview. Based on feedback during the public technical preview, we may decide to make backward-breaking changes to improve security, usability, or performance that could necessitate API changes in our drivers. If this occurs, we will open one or more new RUST tickets to address the changes.

In summary the FLE API is stable, but the QE API is not due to possible changes in the server before its general availability (GA) release. I've closed this GH Issue since [RUST-569](https://jira.mongodb.org/browse/RUST-569) has been closed as fixed. If you have additional questions, please don't hesitate to continue the discussion here.

Sincerely,
James

**Comment by LuisOsta** [comment]
@JamesKovacs Thanks for the clarification! Since the FLE one is stable, would it be possible to not put that in the unstable feature? In https://mongodb.github.io/mongo-rust-driver/manual/encryption.html it mentions that in order to use either you have to enable the unstable in-use encryption feature

**Comment by masterbater** [comment]
> Hi, @LuisOsta,
> 
> Thank you for reaching out to clarify. The API for Field-Level Encryption (FLE) is stable and has been present in our other drivers for years. The API for Queryable Encryption (QE) is new and the server functionality is in public technical preview. Based on feedback during the public technical preview, we may decide to make backward-breaking changes to improve security, usability, or performance that could necessitate API changes in our drivers. If this occurs, we will open one or more new RUST tickets to address the changes.
> 
> In summary the FLE API is stable, but the QE API is not due to possible changes in the server before its general availability (GA) release. I've closed this GH Issue since [RUST-569](https://jira.mongodb.org/browse/RUST-569) has been closed as fixed. If you have additional questions, please don't hesitate to continue the discussion here.
> 
> Sincerely, James

If this already GA, is it possible if you work with prisma to support this feature?
